swift/Plugins/CompileSlice/CompileSlicePlugin.swift

```diff
@@ -22,108 +56,104 @@ enum PluginError: Error {
     }
 }
 
-/// Represents the contents of a `slice-plugin.json` file
-/// - sources: List of Slice files or directories containing Slice files. Can not be empty.
-/// - search_paths: List of directories to add as search paths when calling slice2swift. Can be omitted.
+/// Represents the contents of a `slice-plugin.json` file.
+///
+/// - `sources`: Optional list of Slice files or directories containing Slice files. Paths are relative to the
+///   directory containing the `slice-plugin.json` file. Slice files declared directly in the target's source files
+///   are always included and do not need to be listed here.
+///
+/// - `search_paths`: Optional list of directories to add as `-I` search paths when invoking `slice2swift`.
+///   These paths are also relative to the config file location.
+///
 /// - Example:
-/// ```
+/// ```json
 /// {
 ///     "sources": ["Slice"],
 ///     "search_paths": ["Slice", "../OtherModule/Slice"]
 /// }
 /// ```
 struct Config: Codable {
-    /// List of Slice files or directories containing Slice files
-    var sources: [String]
-    /// List of directories to add as search paths when calling slice2swift.
+    /// Optional list of Slice files or directories containing Slice files.
+    /// Paths are relative to the `slice-plugin.json` file. Slice files in the target's sources are always included.
+    var sources: [String]?
```

Comment:
sources is now optional, to add Slice files that are not in the target inputs. Maybe would be clear to have `additional_sources`.

Reply:
I think we need to restructure this config in general. Maybe do in a follow-up PR?

Reply:
I prefer to do it in a separate PR, as it is not clear what the new structure would be. The main issue fixed here is to allow using the plug-in in Xcode projects.